lib/irb.rb

```diff
@@ -736,7 +736,13 @@ def prompt(prompt, ltype, indent, line_no) # :nodoc:
     end
 
     def output_value # :nodoc:
-      printf @context.return_format, @context.inspect_last_value
+      str = @context.inspect_last_value
+      multiline_p = /\A.*\Z/ !~ str
```

Comment:
This line will occur an exception if `str` is encoded in an encoding not compatible with UTF-8.

Reply:
But, I found that the current implementation also occurs an exception:

```
irb(main):001:0> o = Object.new
irb(main):002:0> def o.inspect; "abcdef".encode("ISO-2022-JP"); end
=> :inspect
irb(main):003:0> o
Traceback (most recent call last):
       16: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/exe/irb:11:in `<top (required)>'
       15: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:398:in `start'
       14: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:469:in `run'
       13: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:469:in `catch'
       12: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:470:in `block in run'
       11: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:535:in `eval_input'
       10: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb/ruby-lex.rb:134:in `each_top_level_statement'
        9: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb/ruby-lex.rb:134:in `catch'
        8: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb/ruby-lex.rb:135:in `block in each_top_level_statement'
        7: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb/ruby-lex.rb:135:in `loop'
        6: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb/ruby-lex.rb:150:in `block (2 levels) in each_top_level_statement'
        5: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:536:in `block in eval_input'
        4: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:693:in `signal_status'
        3: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:540:in `block (2 levels) in eval_input'
        2: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:739:in `output_value'
        1: from /home/mrkn/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/irb-1.2.1/lib/irb.rb:739:in `printf'
Encoding::CompatibilityError (incompatible character encodings: UTF-8 and ISO-2022-JP)
Maybe IRB bug!
```

Reply:
It's the same behavior of old IRB.

Reply:
OK.